package yosysZKP;

message WireValues {
  message Entry {
    required string wirename = 1;
    required bool value = 2; 
  }
  repeated Entry entries = 1;
}

message TruthTableEntry {
  repeated bool inputs = 1;
  repeated bool outputs = 2;
  required bytes nonce = 3;
}

message TruthTable {
  repeated TruthTableEntry entries = 1;
}

message FullState {  
  repeated TruthTable gates = 1;
}


message TableCommitment {
  repeated bytes entryhashes =1;
}

message Commitment {
  repeated TableCommitment gatehashes = 1;
}

message ExecutionReveal {
  required WireValues exec = 1;
  repeated TruthTableEntry entries = 2;
}

message ScramblingReveal {
  required WireValues keys = 1;
  repeated TruthTable gates = 2;
}
